### ISSUE TITLE
Sporadic failure in test ControllerMetricsTest.streamMetricsTest

### ISSUE THREAD

**Issue by RaulGracia** [pravega/pravega#5657]
**Problem description**
Found test `io.pravega.test.integration.ControllerMetricsTest.streamMetricsTest` failing due to:
```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at org.junit.Assert.assertNotNull(Assert.java:722)
	at io.pravega.test.integration.ControllerMetricsTest.streamMetricsTest(ControllerMetricsTest.java:212)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Problem location**
`io.pravega.test.integration.ControllerMetricsTest.streamMetricsTest`

**Suggestions for an improvement**
Check if this is a real bug and make the test to pass reliably.

**Comment by pbelgundi** [comment]
Another occurrence: https://github.com/pravega/pravega/pull/5827/checks?check_run_id=2181302541#step:5:702
```
io.pravega.test.integration.StreamMetricsTest > testRollingTxnMetrics FAILED
    org.junit.runners.model.TestTimedOutException: test timed out after 30000 milliseconds
        at java.base@11.0.10/java.lang.Thread.sleep(Native Method)
        at app//io.pravega.test.common.TestUtils.await(TestUtils.java:77)
        at app//io.pravega.test.common.AssertExtensions.assertEventuallyEquals(AssertExtensions.java:67)
        at app//io.pravega.test.common.AssertExtensions.assertEventuallyEquals(AssertExtensions.java:93)
        at app//io.pravega.test.integration.StreamMetricsTest.testRollingTxnMetrics(StreamMetricsTest.java:372)
```

**Comment by RaulGracia** [comment]
Fixed via #4999.
